### ISSUE TITLE
Usability Improvement: Streamdeck USB instance ID

### ISSUE THREAD

**Issue by matt-cahill** [DCS-Skunkworks/DCSFlightpanels#68]
Each panel is referenced in the profile save file by its USB ID and each streamdeck button is tied to that panel using the same ID in the json config block.

If a user's panel ID changes for whatever reason the user will need to update a large number of entries to fix their profile. As an example the simple master caution 2 button config example provided for #67 contains 9 references. Additionally JSON is very strict which could cause issues for people who don't know what they are doing.

PanelInstanceID provided by the UI and used to reference the panel.
```
\\?\hid#vid_0fd9&pid_006c#6&27db50b8&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
```

In the JSON the backslashes are also escaped which may exacerbate the issue.
```
\\\\?\\hid#vid_0fd9&pid_006c#6&27db50b8&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
```

Would it be possible to reference the panel USB ID once to identify the panel and then map that to some kind of arbitrary panel reference ID to be used in the JSON to tie the buttons to that panel?
That way a user would only have to update a single line to fix their profile.

An example might look something like this
```diff
PanelType=StreamDeck
PanelInstanceID=\\?\hid#vid_0fd9&pid_006c#6&27db50b8&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
+PanelReferenceID=Panel_1
PanelSettingsVersion=2X
BeginPanelJSON
{
  "$type": "System.Collections.Generic.List`1[[NonVisuals.StreamDeck.StreamDeckLayer, NonVisuals]], mscorlib",
  "$values": [
    {
      "$type": "NonVisuals.StreamDeck.StreamDeckLayer, NonVisuals",
      "Name": "Home",
      "StreamDeckButtons": {
        "$type": "System.Collections.Generic.List`1[[NonVisuals.StreamDeck.StreamDeckButton, NonVisuals]], mscorlib",
        "$values": [
          {
            "$type": "NonVisuals.StreamDeck.StreamDeckButton, NonVisuals",
            "StreamDeckButtonName": 1,
            "Face": {
              "$type": "NonVisuals.StreamDeck.FaceTypeImage, NonVisuals",
              "FaceType": 2,
              "ConfigurationOK": true,
              "TextFont": "Consolas, 12pt",
              "FontColor": "Black",
              "BackgroundColor": "Black",
              "ImageFile": "..\\..\\test\\testbouton.png",
              "StreamDeckButtonName": 1,
-             "StreamDeckInstanceId": "\\\\?\\hid#vid_0fd9&pid_006c#6&27db50b8&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}",
+             "StreamDeckReferenceId": "Panel_1",
              "OffsetX": 0,
              "OffsetY": 0
            },
            "ActionForPress": null,
            "ActionForRelease": null,
            "HasConfig": true,
            "ActionType": 0,
            "FaceType": 2
          }
        ]
      },
      "IsVisible": true,
-     "StreamDeckInstanceId": "\\\\?\\hid#vid_0fd9&pid_006c#6&27db50b8&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}",
+     "StreamDeckReferenceId": "Panel_1",
    }
  ]
}
EndPanelJSON
```

**Comment by jdahlblom** [comment]
Yes absolutely. This has been a problem because in my previous class structure a button didn't know anything about the panel it was associated with. I will fix that.

**Comment by jdahlblom** [comment]
I will use a hash based on the USB HID path. It might be that the various objects won't have to export this although they will use it in the background. This way the user won't have to bother with it.

**Comment by matt-cahill** [comment]
This seems fixed but as per this comment https://discordapp.com/channels/533342958712258572/700674160304848906/707491487121670205
it looks like this new hash is being output by the `Copy Panel USB ID` button now (as of v4.0.40) rather than the actual USB ID.

**Comment by matt-cahill** [comment]
fixed in 4.0.52
